### PR TITLE
CON-2326: Search route has display bugs

### PR DESCRIPTION
**Pull request by richardgorman**

### CODE REVIEW

**Comment by coldi** [comment]
When transitioning from a category page to a product page, `SET_VIEW_TITLE` action gets dispatched twice. First time with no title (`title: ''`) and second with the correct product title.
Second dispatch only would be the expected behavior and similar to other page transitions.

**Comment by coldi** [comment]
Filters seem not to work on the search page. Instead I get this:
image

The pipeline error might be temporary but the `Uncaught TypeError` should be solved.

File: components/View/connector.js [hunk 1]
```diff
@@ -6,15 +6,16 @@
...
+import setViewTop from './action-creators/setViewTop';
 
 /**
  * Maps the contents of the state to the component props.
  * @param {Object} state The current application state.
  * @return {Object} The extended component props.
  */
 const mapStateToProps = state => ({
-  viewTop: state.history.state.viewTop,
+  viewTop: state.ui.general.isTop,
```
Comment (coldi) [inline components/View/connector.js]: Please consider to introduce a selector for this. At least something like `getGeneralUI()`.

File: components/View/index.jsx [hunk 2]
```diff
@@ -86,6 +95,37 @@ class View extends Component {
...
+    if (!this.element) {
+      return;
+    }
+
+    const isViewTop = this.element.scrollTop === 0;
+
+    if (isViewTop !== this.props.viewTop) {
+      this.props.setTop(isViewTop);
+    }
+  }, 5);
```
Comment (coldi) [inline components/View/index.jsx]: Please remove this magic number and use some constant instead.
Also, 5ms seems quite low to me. Can't this be a little higher? Maybe something around 50ms?

File: components/Navigator/actions/submitSearch.js [hunk 3]
```diff
@@ -8,64 +8,68 @@
...
     return;
   }
 
-  // Perform search by entering/updating search route
+  // Perform search by entering/updating search route.
   const { history } = state;
   const { sort, ...otherParams } = history.queryParams;
   const prevSearchPhrase = getSearchPhrase(state);
 
+  dispatch(setActiveFilters({}));
```
Comment (coldi) [inline components/Navigator/actions/submitSearch.js]: You may consider dispatching this within a subscription.

File: components/Navigator/actions/submitSearch.js [hunk 4]
```diff
@@ -8,64 +8,68 @@
...
+  dispatch(getSearchResults());
+
+  dispatch(setViewTop(true));
```
Comment (coldi) [inline components/Navigator/actions/submitSearch.js]: You may consider dispatching this within a subscription.

File: components/Navigator/actions/toggleProgressBar.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+import { toggleProgressBar } from '../action-creators';
+import { isProgressBarShowing } from '../selectors';
+
+/**
+ * Toggles the visibility of the navigation progress bar.
+ * @param {boolean} isActive Whether or not to activate the progress bar.
+ * @return {Function} A redux thunk.
+ */
+const setProgressBar = isActive => (dispatch, getState) => {
```
[the remark this answers is not in the document]
Comment (richardgorman) [reply components/Navigator/actions/toggleProgressBar.js]: Corrected.

File: components/View/index.jsx (hunk 2 above)
Comment (richardgorman) [reply components/View/index.jsx]: Done.

File: components/Navigator/actions/submitSearch.js (hunk 3 above)
Comment (richardgorman) [reply components/Navigator/actions/submitSearch.js]: Not in the scope of this ticket.

File: components/Navigator/actions/submitSearch.js (hunk 4 above)
Comment (richardgorman) [reply components/Navigator/actions/submitSearch.js]: Not in the scope of this ticket.

File: components/Navigator/components/Content/components/Title/connector.js [hunk 6]
```diff
@@ -14,7 +14,7 @@ import { connect } from 'react-redux';
  */
 const mapStateToProps = state => ({
   action: state.history.action,
-  title: state.history.state.title || '',
+  title: state.ui.general.title || '',
```
[the remark this answers is not in the document]
Comment (richardgorman) [reply components/Navigator/components/Content/components/Title/connector.js]: Done

File: components/View/connector.js (hunk 1 above)
Comment (richardgorman) [reply components/View/connector.js]: Done

File: components/Navigator/actions/submitSearch.js (hunk 3 above)
Comment (coldi) [reply components/Navigator/actions/submitSearch.js]: Okay

File: components/Navigator/actions/submitSearch.js (hunk 4 above)
Comment (coldi) [reply components/Navigator/actions/submitSearch.js]: Okay

**Pull request merged**